src/leveldb/port/atomic_pointer.h

```diff
@@ -52,13 +50,6 @@ namespace port {
 // http://msdn.microsoft.com/en-us/library/ms684208(v=vs.85).aspx
 #define LEVELDB_HAVE_MEMORY_BARRIER
 
-// Mac OS
-#elif defined(OS_MACOSX)
-inline void MemoryBarrier() {
-  OSMemoryBarrier();
-}
-#define LEVELDB_HAVE_MEMORY_BARRIER
```

Comment:
Not sure about this removal either. Can you expand on the changes? These seem to be quite fundamental.

Reply:
The OSMemoryBarrier is depreciated in MacOS. I replaced the file with and updated version of same file found here:  https://github.com/google/leveldb/blob/f2e7603c417018d5492e931e4a5b75b8fd571c67/port/atomic_pointer.h

It works as shown in the snapshot above. gridcoinresearch vs gridcoin (gridcoin is with the new atomic version)